### ISSUE TITLE
Extraction of coefficients fails for saturated model

### ISSUE THREAD

**Issue by FlorianScharf** [psyphh/lslx#1]
Hi PH,

I am running a simulation study utilizing lslx. The penalty term is chosen by BIC. Everything works well until once every 1000 samples, a low-level error occurs reporting "unsupported type" in a function named:
formatC(rmsea,...) when extracting the coefficients.

Sometimes for this specific sample other selectors work, sometimes they don't.

I traced down the problem - I think. It occurs when a saturated model is chosen and no parameter is shrunken to zero (confirmed with $extract_coefficients). In my case of a multiple regression this results in an invalid value from the rmsea computation which then throws the error.

I am not sure if this is intended behavior (at least the warning is not informative) but theoretically, there's nothing wrong about my model apart from it being saturated.

I attach a minimal example. Hope this is comprehensive.

Best,
Florian 

[Archiv.zip](https://github.com/psyphh/lslx/files/4503289/Archiv.zip)

**Comment by psyphh** [comment]
Hi Florian
Thank you for providing such a clear example.
By the error message, this problem arises when using formatC() for the value of RMSEA, which is NA in this example
Actually, lslx still successfully extracts BIC related results, but cannot print the RMSEA in $summarize() method.
I will fix it soon and take care related problems associated with saturated models.

If you are now conducting simulations and hope to handle this problem as soon as possible, you can try other $extract_xxx() methods.
For example, you can extract asymptotic covariance matrix of coefficients by $extract_coefficient_acov(selector)

Best,
PH

**Comment by psyphh** [comment]
Hi Florian
The bug is fixed (at least for this example).
The updated code is uploaded to GitHub.
I found that the example you provided is not a "standard case" in SEM.
Despite that the model selects all of the covariates, the final shrinkage estimator doesn't yield a "saturated model".
The model-implied matrix is not equal to the sample covariance matrix, and hence the loss function value is larger than zero.
In lslx, loss = 0 will set rmsea = 0 and nnfi = 1.
However, df = 0 but loss > 0 will set both rmsea and nnfi as NaN.

I also found that formatC(NA) will result in an error, but formatC(NaN) will not.
So I simply replace NA by NaN.
Hope that the problem is fixed.

Best,
PH

**Comment by FlorianScharf** [comment]
Hi PH,

thank you for your fast fix for this bug.

You are right that this case is not typical for SEMs (and that it is not saturated in a strict sense due to the penalization)

I can confirm that it works for the example I sent you and it also works in all simulation conditions based on this model (there are no more at the moment - it's "just" a bachelor thesis simulation).

Apart from not throwing errors any more, the output behaves as you described - so this seems to work fine.

Best,
Florian
